core/src/main/scala/org/apache/spark/serializer/JavaSerializer.scala

```diff
@@ -138,7 +142,7 @@ private[spark] class JavaSerializerInstance(
  */
 @DeveloperApi
 class JavaSerializer(conf: SparkConf) extends Serializer with Externalizable {
-  private var counterReset = conf.getInt("spark.serializer.objectStreamReset", 100)
+  private var counterReset = conf.getInt("spark.serializer.objectStreamReset", 3)
```

Comment:
I think 100 is reasonable for normal use cases, if you have a lot of big objects in your case, you should set this conf in your application, instead of changing the default behavior in spark code.